reward/staking_info.go

```diff
@@ -134,26 +134,28 @@ func (s *StakingInfo) GetStakingAmountByNodeId(nodeId common.Address) (uint64, e
 	return s.CouncilStakingAmounts[i], nil
 }
 
-type uint64Slice []uint64
+type float64Slice []float64
 
-func (p uint64Slice) Len() int           { return len(p) }
-func (p uint64Slice) Less(i, j int) bool { return p[i] < p[j] }
-func (p uint64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p float64Slice) Len() int           { return len(p) }
+func (p float64Slice) Less(i, j int) bool { return p[i] < p[j] }
+func (p float64Slice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
 
-func CalcGiniCoefficient(stakingAmount uint64Slice) float64 {
-	sort.Sort(stakingAmount)
+func CalcGiniCoefficient(stakingAmount float64Slice) float64 {
+	tempStakingAmount := make(float64Slice, len(stakingAmount))
+	copy(tempStakingAmount, stakingAmount)
+	sort.Sort(tempStakingAmount)
```

Comment:
What's the reason `copy` is added here?

Reply:
the use of function is changed.
before change, a temp list is given to the input.
but now, original list is given. so copying is needed before sorting